[FLINK-12378][docs] Consolidate FileSystem Documentation

sjwiesman opened the pull request:
## What is the purpose of the change



Currently flink's filesystem documentation is spread across a number of pages without any clear connection. A non-exhaustive list of issues includes:

    S3 documentation spread across many pages
    OSS filesystem is listed under deployments when it is an object store
    deployments/filesystem.md has a lot of unrelated information

We should create a filesystem subsection under deployments with multiple pages containing all relevant information about Flink's filesystem abstraction.

This PR also resolves FLINK-8513 and FLINK-10249 which were minor additions to the S3 documentation. 

## Verifying this change
N/A

## Does this pull request potentially affect one of the following parts:

  - Dependencies (does it add or upgrade a dependency): (yes / **no**)
  - The public API, i.e., is any changed class annotated with `@Public(Evolving)`: (yes / **no**)
  - The serializers: (yes / **no** / don't know)
  - The runtime per-record code paths (performance sensitive): (yes / **no** / don't know)
  - Anything that affects deployment or recovery: JobManager (and its components), Checkpointing, Yarn/Mesos, ZooKeeper: (yes / **no** / don't know)
  - The S3 file system connector: (**yes** / no / don't know)
This does not touch S3 file system code but does touch the documentation. 

## Documentation

  - Does this pull request introduce a new feature? (yes / **no**)
  - If yes, how is the feature documented? (not applicable / **docs** / JavaDocs / not documented)

sjwiesman commented:
@flinkbot attention @fhueske

StephanEwen commented on docs/ops/deployment/aws.md:
```diff
@@ -64,81 +64,11 @@ HADOOP_CONF_DIR=/etc/hadoop/conf ./bin/flink run -m yarn-cluster -yn 1 examples/
 
-This setup is a bit more complex and we recommend using our shaded Hadoop/Presto file systems
-instead (see above) unless required otherwise, e.g. for using S3 as YARN's resource storage dir
+Apache Flink provides native [S3 FileSystem's](../filesystems/s3.html) out of the box and we recomend using them unless required otherwise, e.g. for using S3 as YARN's resource storage dir
```
recomend --> recommend

StephanEwen replied:
"native" --> "built-in" ?

StephanEwen replied:
I am also wondering if we should simply drop the section below, about Hadoop's S3 file systems.
We can mention that Flink also supports Hadoop's file systems and refer to Hadoop docs for details.

sjwiesman replied:
I think that's reasonable. I believe most users are using the built-in S3 filesystems at this point.

StephanEwen commented on docs/ops/filesystems/index.md:
```diff
@@ -0,0 +1,83 @@
+under the License.
+-->
+
+Apache Flink uses to consume and persistently store data, both for results of applications and for fault tolerance and recovery.
```
Apache Flink uses *file systems* ?

StephanEwen commented on docs/ops/filesystems/s3.md:
```diff
@@ -0,0 +1,134 @@
+
+Note that these examples are *not* exhaustive and you can use S3 in other places as well, including your [high availability setup](../jobmanager_high_availability.html) or the [RocksDBStateBackend]({{ site.baseurl }}/ops/state/state_backends.html#the-rocksdbstatebackend); everywhere that Flink expects a FileSystem URI.
+
+For most use cases, you may use one of our shaded `flink-s3-fs-hadoop` and `flink-s3-fs-presto` S3filesystem wrappers which are self-contained and easy to set up.
```
S3 filesystem

StephanEwen commented on docs/ops/filesystems/s3.md:
```diff
@@ -0,0 +1,134 @@
+You can use S3 objects like regular files by specifying paths in the following format:
+
+{% highlight plain %}
+s3://<your-bucket>/<endpoint>
```
<endpoint> --> <key> ?

sjwiesman commented:
Thanks for the review, I consolidated on "file systems" and removed the Hadoop references from AWS and OSS pages and replaced it with an extra section about configuring Hadoop on filesystems/index.md.
